torch/csrc/jit/script/compiler.cpp

```diff
@@ -1349,6 +1377,19 @@ struct to_ir {
       }
     }
 
+    // for (el in my_list):
+    if (itrs[0].kind() == TK_VAR) {
+      auto listArg = environment_stack->getSugaredVar(Var(itrs[0]).name());
```

Comment:
This breaks for something like `for item in [1, 2, 3]:`, instead of using the `environment_stack` you should be able to just use the sugared value for the iterator